File: 45-jump-game-ii/jump-game-ii.ts
function jump(nums: number[]): number {
    let l = 0;
    let r = 0;
    let jumpCount = 0;

    while(r < nums.length - 1){
        let curFarthest = 0;
        for(let i = l; i <= r; i ++) {
            curFarthest = Math.max(curFarthest,i + nums[i]);
        }
        l = r + 1;
        r = curFarthest;
        jumpCount ++;
    }
    return jumpCount;
};

File: 45-jump-game-ii/jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int l = 0;
        int r = 0;
        int count = 0;
        while ( r < nums.length - 1) {
            int curFar = 0;
            for(int i = l; i <= r ; i ++) {
                curFar = Math.max(curFar , i + nums[i]);
            }
            l = r + 1;
            r = curFar;
            count ++;
        }
        return count;
    }
}